### DISCUSSION TITLE
How to build with external submodules

### DISCUSSION THREAD

**Discussion by andrew11221** [microsoft/onnxruntime#8157] [Other Q&A] [answered]
If I have some of the submodules prebuilt (ex: eigen, flatbuffers, etc) as .so libs, how could I build ONNXRuntime using them instead of what's in cmake/external?

Am I correct in thinking that I don't need sources if I already have the .so built?

**Comment by snnn** [comment, marked as answer]
It's not supported.

**Reply by andrew11221** [reply]
I see. How hard do you think it would be to support it and do you have any tips?

**Reply by snnn** [reply]
I guess maybe you are asking for this build option: https://github.com/microsoft/onnxruntime/blob/master/cmake/CMakeLists.txt#L103 But it is an incomplete work.

**Reply by andrew11221** [reply]
Ok yes I started looking into this build option earlier but was having some issues. I'll fiddle with it and make it work, thanks!

**Reply by andrew11221** [reply]
As a follow-up, are all submodules in cmake/external needed if I only want to do CPU inference?
